viper/src/main/scala/viper/api/SilverImplementation.scala

```diff
@@ -67,7 +67,37 @@ class SilverImplementation[O](o:OriginFactory[O])
               prog.methods.asScala.toList, Seq(/* no extension members */))()
               
     //println("=============\n" + program + "\n=============\n")
-    
+
+    val consistencyErrors = program.checkTransitively
+
+    if(consistencyErrors.nonEmpty) {
+      Warning("These errors may indicate a bug in VerCors:")
+      consistencyErrors.foreach(Warning("%s", _))
+    }
+
+    val sugarErrors = SilverTreeCompare.syntacticSugarIssues(program)
```

Comment:
This is run by default. It indeed checks for encoding problems, in particular it will catch issues related to what viper calls syntactic sugar in their frontend. E.g. the issue we had before, I think with misshaped forall*-s including `?:` or so. These should be consistency errors. When they are not caught as consistency errors, reparsing the AST will lead to differences (because the syntactic sugar will be desugared), indicating a likely bug in viper.